### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by Tracy-miranja**
- [ ] Remove current_user method in ApplicationController, devise will provide us with one.
- [ ] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- [ ] Can register a new user.
- [ ] User logs in with a combination of email and password.
- [ ] Hashed password should be stored in the database.
- [ ] Ask for confirmation of email.
- [ ] Can reset password.
- [ ] [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (FrederickMih) [inline app/models/user.rb]: - [x] As requested in the requirements, let's kindly make sure the application asks for the user to confirm email hence you'll be required to enable` :confirmation` in the user model.

File: app/views/users/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "users/shared/links" %>
```
Comment (FrederickMih) [inline app/views/users/registrations/new.html.erb]: - [x] You're equally requested to modify the user's registration and login page(views) to match the wireframe provided in the project requirements.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @Tracy-miranja, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights :green_circle: 

- Devise had been configured to provide the app with some level of security :+1: I was able to securely register and login. :+1: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉
### Highlights
- Correct workflow was used ✔️
- Good project structure ✔️
- Well-organized and clean project ✔
- Good commit history ✔️
thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**